content/chainguard/administration/sdk-authentication.md

```diff
@@ -0,0 +1,203 @@
+}
+```
+
+The example returns the image's digest and the OCI manifest data. This proves that you've gotten a response back from the registry and that authentication worked as expected.
+
+This example doesn't really reflect a real-world use case, as users will generally access `cgr.dev/chainguard` container repositories without authenticating. However, by presenting a token it will trigger the authentication checks, making this example useful for illustrating how this can be done with the Chainguard SDK.
+
+You can experiment with updating this example to retrieve information about a different Chainguard container image by changing the `cgr.dev/chainguard/static` value in the `main()` function. You could replace this with any Chainguard repository you have access to, and the example will return information about that image.
+
+
+## Example 2 — Authenticate with an Assumed Identity 
+
+The [`exchange` example](https://github.com/chainguard-dev/sdk/tree/main/examples/registry/exchange) demonstrates how to exchange a token for an assumed identity to access the registry.
+
+This example consists of the following Go code in a `main.go` file (with comments removed):
+
+```go
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"os"
+
+	"chainguard.dev/sdk/auth"
+	"chainguard.dev/sdk/auth/ggcr"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+const (
+	sub = "720909c9f5279097d847ad02a2f24ba8f59de36a/a033a6fabe0bfa0d"
+)
+
+func main() {
+	ctx := context.Background()
+	ts := auth.NewChainctlTokenSource(ctx)
+
+	desc, err := remote.Get(name.MustParseReference("cgr.dev/chainguard/static"), remote.WithAuthFromKeychain(ggcr.Keychain(sub, ts)))
+	if err != nil {
+    	log.Fatalf("error getting reference: %v", err)
+	}
+	enc := json.NewEncoder(os.Stdout)
+	enc.SetIndent("", "  ")
+	_ = enc.Encode(desc)
+}
+```
+
+This example is similar to the previus one, but has the following differences:
+
+* It creates a constant named `sub`. In this example, the `sub` constant's value is set to the UIDP of a Chainguard identity named `all-users` which can be assumed by any Chainguard user.
+* It takes the token retrieved with the `chainctl` binary and exchanges it for the assumable identity to make a call to the Chainguard registry in order retrieve some information about the supplied image. The example uses the `cgr.dev/chainguard/static` Starter image.
```

Comment:
I had a hard time seeing the differences, and I'm still a little unsure. Worth calling out a bit more maybe (e.g. ggcr.Keychain call)